Highway.cpp

```diff
@@ -13,17 +17,44 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
+    // assert(false);
+    if (Car* car = dynamic_cast<Car*>(v))
+    {
+        // car->setSpeed(speedLimit);
+        car->closeWindows();
+    }
 
+    if (Motorcycle* mc = dynamic_cast<Motorcycle*>(v))
+    {
+        mc->lanesplitAndRace();
+    }
+
+    if (SemiTruck* st = dynamic_cast<SemiTruck*>(v))
+    {
+        st->speedUp();
+    }
     /*
     depending on the derived type, call the member function that doesn't evade the cops. 
     */
 }
 
 void Highway::removeVehicleInternal(Vehicle* v)
 {
-    assert(false);
+    // assert(false);
+    if (Car* car = dynamic_cast<Car*>(v))
+    {
+        car->tryToEvade();
+    }
+
+    if (Motorcycle* mc = dynamic_cast<Motorcycle*>(v))
+    {
+        mc->tryToEvade();
+    }
 
+    if (SemiTruck* st = dynamic_cast<SemiTruck*>(v))
+    {
+        st->tryToEvade(); // this is overridden
+    }
```

Comment:
use `else if`
When you don't use it, you're still checking if its a motorcycle or truck, even if it's a car.
- [ ] revise